### PR TITLE
Fix block generation by updating codec versions in wasm

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

**Comment by arkpar** [comment]
`cargo update -p parity-codec` would have done the same effectively

**Comment by arkpar** [comment]
Some tests still need updating

**Comment by gavofyork** [comment]
there were instances of `parity-codec-derive = { version = "~1.0" }` before. pretty sure this was the issue. the rest of the version updates was for consistency.

File: node/service/src/chain_spec.rs [hunk 1]
```diff
@@ -39,72 +39,81 @@ fn staging_testnet_config_genesis() -> GenesisConfig {
...
-
-			cooloff_period: 12 * 60 * 24 * 4, // 4 day cooling off period if council member vetoes a proposal.
-			voting_period: 12 * 60 * 24, // 1 day voting period for council members.
+			candidacy_bond: 10 * DOLLARS,
+			voter_bond: 1 * DOLLARS,
+			present_slash_per_voter: 1 * CENTS,
+			carry_count: 6,
+			presentation_duration: 1 * DAYS,
+			approval_voting_period: 2 * DAYS,
+			term_duration: 28 * DAYS,
```
Comment (bkchr) [inline node/service/src/chain_spec.rs]: Is that intentionally that `term_duration` changes from 24 to 28 days?

Comment (gavofyork) [reply node/service/src/chain_spec.rs]: yeah

File: core/misbehavior-check/Cargo.toml [hunk 2]
```diff
@@ -4,10 +4,10 @@ version = "0.1.0"
 authors = ["Parity Technologies <admin@parity.io>"]
 
 [dependencies]
-parity-codec = { version = "2.0", default-features = false }
-substrate-primitives = { path = "../primitives", default-features = false }
-sr-primitives = { path = "../sr-primitives", default-features = false }
-sr-io = { path = "../sr-io", default-features = false }
+parity-codec = { version = "~2.0.1", default_features = false }
```
Comment (tomusdrw) [inline core/misbehavior-check/Cargo.toml]: Why is `default_features` changed?

File: core/bft/Cargo.toml [hunk 3]
```diff
@@ -5,7 +5,7 @@ authors = ["Parity Technologies <admin@parity.io>"]
 
 [dependencies]
 futures = "0.1.17"
-parity-codec = { version = "2.0" }
+parity-codec = "~2.0.1"
```
Comment (tomusdrw) [inline core/bft/Cargo.toml]: Why `~` and bugfix version? Can't we manage the exact version in the `Cargo.lock`?

**Pull request merged**